src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -467,6 +436,9 @@ await SetChildResourceAsync(child, state, startTimeStamp, stopTimeStamp)
 
     private async Task PublishResourcesInitialStateAsync(CancellationToken cancellationToken)
     {
+        // Initialize all parameter resources up front
+        await _parameterProcessor.InitializeParametersAsync(_model.Resources.OfType<ParameterResource>()).ConfigureAwait(false);
```

Comment:
I think we should write this as if in the future that parmeters might be added to the model late instead of resolving them in this call just have the ParameterProcessor find the parameters in the app model and if they are unresolved then added them to the set to be prompted.

Reply:
I abandoned that idea, the code was too complex. You never know when you are done and the next prompt might have more values. Felt pretty strange from a UX pov. I think it would be possible to rewrite this to do that but we have no scenario for it that works in practice.